### PR TITLE
BE-889 Adding reasons why commit/commit_files are excluded

### PR DESCRIPTION
**Pull request by josecordaz**
https://pinpt-hq.atlassian.net/browse/BE-889

**Description:**

Adding reasons why commit/commit_files are excluded in the metadata field.

### CODE REVIEW

File: pkg/git/commitactivity.go [hunk 1]
```diff
@@ -170,11 +183,23 @@ func GenerateCommitActivity(ctx context.Context, logger log.Logger, dir string,
 			if blame.Language == "" {
 				blame.Language = unknownLanguage
 			}
-			excluded := blame.Skipped || cf.Binary || excludedLanguages[blame.Language]
+			skipedReason := blame.Skipped
```
Comment (jhaynie) [inline pkg/git/commitactivity.go]: `skipedReason` -> `skippedReason`

File: pkg/git/commitactivity.go [hunk 2]
```diff
@@ -170,11 +183,23 @@ func GenerateCommitActivity(ctx context.Context, logger log.Logger, dir string,
...
+			isBinary := cf.Binary
+			excByLanguage := excludedLanguages[blame.Language]
+			excluded := skipedReason != "" || isBinary || excByLanguage
 			if excluded {
 				// track our excludes
 				excludedFiles[blame.Commit.SHA] = excludedFiles[blame.Commit.SHA] + 1
+				mtdt, err = getExcludedReason(skipedReason, isBinary, excByLanguage)
+				if err != nil {
+					errors <- err
+				}
```
Comment (jhaynie) [inline pkg/git/commitactivity.go]: probably need a return if an error

File: pkg/git/commitactivity.go [hunk 3]
```diff
@@ -266,6 +296,50 @@ func GenerateCommitActivity(ctx context.Context, logger log.Logger, dir string,
...
+	if skipedRsn != "" {
+		return getJSONStrReason(skipedRsn)
+	}
+	if isBinary {
+		return getJSONStrReason("it is a binary file")
+	} else if excByLanguage {
+		return getJSONStrReason("excluded by language")
+	} else {
+		return getJSONStrReason("unknow reason")
+	}
```
Comment (jhaynie) [inline pkg/git/commitactivity.go]: `unknow` -> `unknown`

File: pkg/git/commitactivity.go [hunk 4]
```diff
@@ -266,6 +296,50 @@ func GenerateCommitActivity(ctx context.Context, logger log.Logger, dir string,
...
+}
+
+func getJSONStrReason(reason string) (*string, error) {
+	t := new(exclusion)
+	t.ExcludeReason = reason
+	b, err := json.Marshal(t)
+	if err != nil {
+		return toStringPointer(""), err
```
Comment (jhaynie) [inline pkg/git/commitactivity.go]: `return nil, err`

File: pkg/git/commitactivity.go (hunk 1 above)
Comment (josecordaz) [reply pkg/git/commitactivity.go]: done

File: pkg/git/commitactivity.go (hunk 2 above)
Comment (josecordaz) [reply pkg/git/commitactivity.go]: done

File: pkg/git/commitactivity.go (hunk 3 above)
Comment (josecordaz) [reply pkg/git/commitactivity.go]: done

File: pkg/git/commitactivity.go (hunk 4 above)
Comment (josecordaz) [reply pkg/git/commitactivity.go]: done

**Comment by jhaynie** [comment]
LGTM 👍

**Comment by jhaynie** [comment]
Need to do a ripsrc release to also pull in those changes

**Comment by jhaynie** [comment]
2.0.0 is released

**Comment by josecordaz** [comment]
thanks

**Pull request merged**